starter-code/pizza.js

```diff
@@ -1,85 +1,200 @@
+		if (state.mushrooms) {
+			$mush.style.visibility = "visible";
+		} else {
+			$mush.style.visibility = "hidden";
+		}
+	});
 }
 
 function renderGreenPeppers() {
-  // Iteration 1: set the visibility of `<section class="green-pepper">`
+	// Iteration 1: set the visibility of `<section class="green-pepper">`
+	document.querySelectorAll(".green-pepper").forEach(function($green) {
+		if (state.greenPeppers) {
+			$green.style.visibility = "visible";
+		} else {
+			$green.style.visibility = "hidden";
+		}
+	});
 }
 
 function renderWhiteSauce() {
-  // Iteration 2: add/remove the class "sauce-white" of `<section class="sauce">`
+	// Iteration 2: add/remove the class "sauce-white" of `<section class="sauce">`
+	document.querySelectorAll(".sauce").forEach(function($whiteSauce) {
+		if (state.whiteSauce) {
+			$whiteSauce.classList.add("sauce-white");
+		} else {
+			$whiteSauce.classList.remove("sauce-white");
+		}
+	});
 }
 
 function renderGlutenFreeCrust() {
-  // Iteration 2: add/remove the class "crust-gluten-free" of `<section class="crust">`
+	// Iteration 2: add/remove the class "crust-gluten-free" of `<section class="crust">`
+	document.querySelectorAll(".crust").forEach(function($glutenFreeCrust) {
+		if (state.glutenFreeCrust) {
+			$glutenFreeCrust.classList.add("crust-gluten-free");
+		} else {
+			$glutenFreeCrust.classList.remove("crust-gluten-free");
+		}
+	});
 }
 
 function renderButtons() {
-  // Iteration 3: add/remove the class "active" of each `<button class="btn">`
+	// Iteration 3: add/remove the class "active" of each `<button class="btn">`
+
+	if (!state.pepperonni) {
+		let element = document.querySelector(".btn-pepperonni"); // btn-mushrooms active");
+		element.classList.remove("active");
+	} else {
+		let element = document.querySelector(".btn-pepperonni");
+		element.classList.add("active");
+	}
+
+	if (!state.mushrooms) {
+		let element = document.querySelector(".btn-mushrooms"); // btn-mushrooms active");
+		element.classList.remove("active");
+	} else {
+		let element = document.querySelector(".btn-mushrooms");
+		element.classList.add("active");
+	}
+
+	if (!state.greenPeppers) {
+		let element = document.querySelector(".btn-green-peppers"); // btn-mushrooms active");
+		element.classList.remove("active");
+	} else {
+		let element = document.querySelector(".btn-green-peppers");
+		element.classList.add("active");
+	}
+
+	if (!state.whiteSauce) {
+		let element = document.querySelector(".btn-sauce"); // btn-mushrooms active");
+		element.classList.remove("active");
+	} else {
+		let element = document.querySelector(".btn-sauce");
+		element.classList.add("active");
+	}
+
+	if (!state.glutenFreeCrust) {
+		let element = document.querySelector(".btn-crust"); // btn-mushrooms active");
+		element.classList.remove("active");
+	} else {
+		let element = document.querySelector(".btn-crust");
+		element.classList.add("active");
+	}
 }
 
 function renderPrice() {
```

Comment:
When you click the buttons on the side you should also update the list of ingredients on your pizza. If you click away pepperoni, then you need to remove "$1 pepperoni" on the side menu. This can be done for instance by creating a string that contains all the text in the menu and then just apply it to the menu with .innerHTML
let string = ""
if(state.pepperonni){string += "$1 Pepperonni"
} else if (state.mushrooms){string += "$1 Mushrooms" ....
and then just set the innerHTML as this string in the end.